http_test.go

```diff
@@ -90,20 +92,20 @@ func TestHttp(t *testing.T) {
 	}
 
 	v := <-api.ch
-	if err := json.NewDecoder(strings.NewReader(v)).Decode(&content); err != nil {
+	if err := json.Unmarshal([]byte(v), &content); err != nil {
 		t.Fatalf("unable to decode content, %v", err)
 	}
 
-	if expected := "www.example.com"; expected != content.Name {
-		t.Errorf("want %v; got %v", expected, content.Name)
+	if want := host; want != content.Name {
```

Comment:
You can instead do:

if got, want := content.Name, host; got != want {
          t.Errorf("Got %v; want %v", got, want)

Always print got first, want second.